src/rtf-plugins/fixture-managers/yarpserver/yarpserverfixture.cpp

```diff
@@ -12,34 +12,19 @@
 #include <rtf/dll/Plugin.h>
 #include <yarp/serversql/yarpserversql.h>
 #include <yarp/os/LogStream.h>
+#include <yarp/os/impl/NameClient.h>
+#include <cmath>
 
 using namespace yarp::os;
 using namespace RTF;
-
-class YarpServerThread : public yarp::os::Thread
-{
-public:
-    void configure(int inArgc, char** inArgv)
-    {
-        argc = inArgc;
-        argv = inArgv;
-    }
-private:
-    int    argc;
-    char** argv{0};
-    virtual void run() override
-    {
-        int ret = yarpserver_main(argc, argv);
-        YARP_UNUSED(ret);
-    }
-};
+//using namespace yarp::name;
 
 class YarpNameServer : public FixtureManager
 {
 public:
     yarp::os::Network net;
     YarpNameServer() {};
-    virtual ~YarpNameServer() {};
+    virtual ~YarpNameServer() {yServer.stop();};
```

Comment:
Why don't we keep this blocking stop in the teardown?

Reply:
+1

Reply:
for what i remember it is better this to be the very last step. rtf calls all the teardowns and then it destroy all the objects.. so if we want the yarpserver shutting down to to be alive during all the other fixture disruption it it better to leave this as it is.

Reply:
If the order of teardown of the fixtures in RTF is reversed w.r.t. to the order of setup, this should not be a problem, as all the fixture using yarp need to be setup after the yarpserver, and so they will be teardown before the yarpserver. This should be the case given https://github.com/robotology/robot-testing/pull/69/files .